doc/user/content/sql/begin.md

```diff
@@ -1,55 +1,166 @@
 ---
 title: "BEGIN"
+description: "`BEGIN` starts a transaction block."
 menu:
   main:
     parent: "commands"
 ---
 
-`BEGIN` starts a transaction block.
+{{% txns/txn-details %}}
+
+Materialize only supports [**read-only** transactions](#read-only-transactions)
+or [**write-only** (specifically, insert-only)
+transactions](#write-only-transactions). See [Details](#details) for more
+information.
 
 ## Syntax
 
-{{< diagram "begin.svg" >}}
+```mzsql
+BEGIN [ <option>, ... ];
+```
 
-Supported `transaction_mode` option values:
+You can specify the following optional settings for `BEGIN`:
 
-Value | Description
-------|----------
-`ISOLATION LEVEL <level>` | Sets the transaction [isolation level](/overview/isolation-level).
-`READ ONLY` | Limits the transaction to read-only operations.
+Option | Description
+-------|----------
+`ISOLATION LEVEL <level>` | *Optional*. If specified, sets the transaction [isolation level](/get-started/isolation-level).
+`READ ONLY` | <a name="begin-option-read-only"></a> *Optional*. If specified, restricts the transaction to read-only operations. If unspecified, Materialize restricts the transaction to read-only or insert-only operations based on the first statement in the transaction.
 
 ## Details
 
-`BEGIN` starts a transaction block.
-All statements in a transaction block will be executed in a single transaction until an explicit [`COMMIT`](/sql/commit) or [`ROLLBACK`](/sql/rollback) is given.
+Transactions in Materialize are either [**read-only**
```

Comment:
This detail section is more on transactions than begin (which is why the examples are included in sections instead of having a separate examples section).  If/when we have a page on that, can do a bit of reorg then.